quake-terminal@diegodario88.github.io/quake-mode.js

```diff
@@ -343,15 +343,20 @@ export const QuakeMode = class {
 			return;
 		}
 
-		let mainMonitorScreen = this._settings.get_int("monitor-screen");
-		const maxNumberOfMonitors = global.display.get_n_monitors() - 1;
-
-		if (mainMonitorScreen > maxNumberOfMonitors) {
-			mainMonitorScreen = maxNumberOfMonitors;
+		// get the monitor index on which the terminal will be rendered
+		const monitorIndex = this._settings.get_boolean("render-on-current-monitor")
+			? global.display.get_current_monitor()
+			: this._settings.get_boolean("render-on-primary-monitor")
+				? Main.layoutManager.primaryIndex
+				: this._settings.get_int("monitor-screen");
```

Comment:
I would suggest avoiding nested ternary operators if possible, as they add complexity to the code and make it harder to read.